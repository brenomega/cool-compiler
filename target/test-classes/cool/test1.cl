-- simple line comment
(* simple block comment *)

cLasS Main inherits IO {
    x : Int <- 10;  -- simple assign
    y : String <- "Hello, COOL!\nNew line";

    (* Block comment
       new line (* nested comment *)
       end *)
    z : Bool <- true;

    if x <= 100 then
        y <- "Greater than 100";
    else
        y <- "Less or equal 100";
    fi;

    while 0 < x loop
        x <- x - 1;
    pool;
}