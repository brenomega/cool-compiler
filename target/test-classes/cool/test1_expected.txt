Line 1: COMMENT('-- simple line comment')
Line 2: COMMENT('(* simple block comment *)')
Line 4: CLASS('class')
Line 4: WHITESPACE(' ')
Line 4: TYPEID('Main')
Line 4: WHITESPACE(' ')
Line 4: INHERITS('inherits')
Line 4: WHITESPACE(' ')
Line 4: TYPEID('IO')
Line 4: WHITESPACE(' ')
Line 4: LBRACE('{')
Line 5: WHITESPACE(' ')
Line 5: OBJECTID('x')
Line 5: WHITESPACE(' ')
Line 5: COLON(':')
Line 5: WHITESPACE(' ')
Line 5: TYPEID('Int')
Line 5: WHITESPACE(' ')
Line 5: ASSIGN('<-')
Line 5: WHITESPACE(' ')
Line 5: INT('10')
Line 5: SEMI(';')
Line 5: WHITESPACE(' ')
Line 5: COMMENT('-- simple assign')
Line 6: WHITESPACE(' ')
Line 6: OBJECTID('y')
Line 6: WHITESPACE(' ')
Line 6: COLON(':')
Line 6: WHITESPACE(' ')
Line 6: TYPEID('String')
Line 6: WHITESPACE(' ')
Line 6: ASSIGN('<-')
Line 6: WHITESPACE(' ')
Line 6: STRING('Hello, COOL! New line')
Line 6: SEMI(';')
Line 8: WHITESPACE(' ')
Line 10: COMMENT('(* Block comment new line (* nested comment *) end *)')
Line 11: WHITESPACE(' ')
Line 11: OBJECTID('z')
Line 11: WHITESPACE(' ')
Line 11: COLON(':')
Line 11: WHITESPACE(' ')
Line 11: TYPEID('Bool')
Line 11: WHITESPACE(' ')
Line 11: ASSIGN('<-')
Line 11: WHITESPACE(' ')
Line 11: TRUE('true')
Line 11: SEMI(';')
Line 13: WHITESPACE(' ')
Line 13: IF('if')
Line 13: WHITESPACE(' ')
Line 13: OBJECTID('x')
Line 13: WHITESPACE(' ')
Line 13: LE('<=')
Line 13: WHITESPACE(' ')
Line 13: INT('100')
Line 13: WHITESPACE(' ')
Line 13: THEN('then')
Line 14: WHITESPACE(' ')
Line 14: OBJECTID('y')
Line 14: WHITESPACE(' ')
Line 14: ASSIGN('<-')
Line 14: WHITESPACE(' ')
Line 14: STRING('Greater than 100')
Line 14: SEMI(';')
Line 15: WHITESPACE(' ')
Line 15: ELSE('else')
Line 16: WHITESPACE(' ')
Line 16: OBJECTID('y')
Line 16: WHITESPACE(' ')
Line 16: ASSIGN('<-')
Line 16: WHITESPACE(' ')
Line 16: STRING('Less or equal 100')
Line 16: SEMI(';')
Line 17: WHITESPACE(' ')
Line 17: FI('fi')
Line 17: SEMI(';')
Line 19: WHITESPACE(' ')
Line 19: WHILE('while')
Line 19: WHITESPACE(' ')
Line 19: INT('0')
Line 19: WHITESPACE(' ')
Line 19: LT('<')
Line 19: WHITESPACE(' ')
Line 19: OBJECTID('x')
Line 19: WHITESPACE(' ')
Line 19: LOOP('loop')
Line 20: WHITESPACE(' ')
Line 20: OBJECTID('x')
Line 20: WHITESPACE(' ')
Line 20: ASSIGN('<-')
Line 20: WHITESPACE(' ')
Line 20: OBJECTID('x')
Line 20: WHITESPACE(' ')
Line 20: MINUS('-')
Line 20: WHITESPACE(' ')
Line 20: INT('1')
Line 20: SEMI(';')
Line 21: WHITESPACE(' ')
Line 21: POOL('pool')
Line 21: SEMI(';')
Line 22: RBRACE('}')